### PR TITLE
MCP-2085 Mock BIP service

### PR DESCRIPTION
**Pull request by dimitri-amida**
## What was the problem?

Integration with the actual BIP API is taking a long time to complete and will probably be impossible to test in local and test environments. In order to complete the remaining routes, we are creating a Mock service mimicking some expected responses from BIP. 

NOT in scope:   [Upload PDF](https://amida.atlassian.net/browse/MCP-2090)

Associated tickets or Slack threads:

- [MCP-2085](https://amida.atlassian.net/browse/MCP-2085)

## How does this fix it?[^1]
- Create a Mock service that mimics BIP responses for some specific collection IDs (see testing step for details)
- Completed the "Complete Processing" route as described in https://github.com/department-of-veterans-affairs/abd-vro/wiki/VRO-v2-Roadmap

## How to test this PR
- Collection ID = 350 with defaults: Processing Goes through until "order exam" step. Then fails because of MAS
- CollectionID = 350 with disabilityActionType": "DECREASE": Return "Not in scope"
- CollectionID = 351: Return "Missing anchor" (wrong station)
- CollectionID = 352: Return "Missing anchor" (missing special issue)
- CollectionID = 353: Does not complete processing after number of attempts is exhausted

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Comment by au2** [comment]
@dimitri-amida please add description to the pr. not everybody in the greater vro team has access to Jira.

**Comment by dimitri-amida** [comment]
@au2 this is still  WIP. I add the details as a last step, when I know what the resolution is and how to test it.

File: controller/src/main/java/gov/va/vro/controller/MasController.java [hunk 1]
```diff
@@ -17,28 +15,14 @@
...
-
-  private final MasConfig masConfig;
+  private final MasProcessingService masProcessingService;
 
   /** Initiate MAS integration. */
   @Override
   public ResponseEntity<MasResponse> automatedClaim(MasAutomatedClaimPayload payload) {
     log.info(
         "Received MAS automated claim request with collection ID {}", payload.getCollectionId());
-    String message;
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/MasController.java:29]: moved all logic to the service layer

File: service/provider/src/main/java/gov/va/vro/service/provider/bip/service/IBipApiService.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import gov.va.vro.model.bip.*;
+import gov.va.vro.service.provider.bip.BipException;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+/** @author warren @Date 11/8/22 */
+public interface IBipApiService {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/bip/service/IBipApiService.java:11]: I only kept the pieces of the API that we use

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationProcessors.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import java.util.List;
+import java.util.function.Function;
+
+/** Helper processors for Mas Integration */
+@Slf4j
+public class MasIntegrationProcessors {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationProcessors.java:21]: extracted this helper class to reduce the size of MasIntegrationRoutes and hopefully make it more readable

File: service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationRoutes.java [hunk 4]
```diff
@@ -173,7 +148,19 @@ private void configureOrderExamStatus() {
     from(ENDPOINT_EXAM_ORDER_STATUS).routeId(routeId).log("Invoked " + routeId);
   }
 
-  private void configureAuditing() {
+  private void configureCompleteProcessing() {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/camel/MasIntegrationRoutes.java:151]: These are the "complete processing" routes

File: service/provider/src/main/java/gov/va/vro/service/provider/mas/service/MasCollectionService.java [hunk 5]
```diff
@@ -123,24 +122,4 @@ private static <T> List<T> merge(List<T> list1, List<T> list2) {
     }
     return new ArrayList<>(result);
   }
-
-  public static GeneratePdfPayload getGeneratePdfPayload(MasTransferObject transferObject) {
```
Comment (dimitri-amida) [inline service/provider/src/main/java/gov/va/vro/service/provider/mas/service/MasCollectionService.java:127]: Moved to helper class

**Review by wlin2019 — APPROVED** [review]
LGTM

**Review by rajesh-Krishnasamy — APPROVED** [review]

**Pull request merged**